src/main/java/com/conveyal/gtfs/PatternFinder.java

```diff
@@ -208,50 +210,53 @@ public static void renamePatterns(
 
     /**
      * Using the 'unique stop id' return the object it actually relates to. Under flex, a stop id can either be a stop,
-     * location or location group, this method decides which.
+     * location or stop area, this method decides which.
      */
     private static Object getStopType(
         String uniqueEntityId,
         Map<String, Stop> stopById,
         Map<String, Location> locationById,
-        Map<String, LocationGroup> locationGroupById
+        Map<String, StopArea> stopAreaById
     ) {
         if (stopById.get(uniqueEntityId) != null) {
             return stopById.get(uniqueEntityId);
         } else if (locationById.get(uniqueEntityId) != null) {
             return locationById.get(uniqueEntityId);
-        } else if (locationGroupById.get(uniqueEntityId) != null) {
-            return locationGroupById.get(uniqueEntityId);
+        } else if (stopAreaById.get(uniqueEntityId) != null) {
+            return stopAreaById.get(uniqueEntityId);
         } else {
             return null;
         }
     }
 
     /**
-     * Extract the 'stop name' from either a stop, location or location group depending on the entity type.
+     * Extract the 'stop name' from either a stop, location or area (via stop area) depending on the entity type.
      */
-    private static String getStopName(Object entity) {
+    private static String getStopName(Object entity, Map<String, Area> areaById) {
         if (entity != null) {
             if (entity instanceof Stop) {
                 return ((Stop) entity).stop_name;
             } else if (entity instanceof Location) {
                 return ((Location) entity).stop_name;
-            } else if (entity instanceof LocationGroup) {
-                return ((LocationGroup) entity).location_group_name;
+            } else if (entity instanceof StopArea) {
+                StopArea stopArea = (StopArea) entity;
+                Area area = areaById.get(stopArea.area_id);
+                return (area != null) ? area.area_name : "stopNameUnknown";
```

Comment:
Why do we have a second stopNameUnknown return here? Can't we use the existing fallback

Reply:
yeah we can, updated.